holoviews/tests/operation/test_datashader.py

```diff
@@ -274,18 +274,37 @@ def test_aggregate_ndoverlay_count_cat_datetimes_microsecond_timebase(self):
         curve = Curve((dates, [1, 2, 3]))
         curve2 = Curve((dates, [3, 2, 1]))
         ndoverlay = NdOverlay({0: curve, 1: curve2}, 'Cat')
-        imgs = aggregate(ndoverlay, aggregator=ds.count_cat('Cat'), width=2, height=2,
+        img = aggregate(ndoverlay, aggregator=ds.count_cat('Cat'), width=2, height=2,
                          x_range=(xstart, xend), dynamic=False)
-        bounds = (np.datetime64('2015-12-31T23:59:59.723518'), 1.0,
-                  np.datetime64('2016-01-03T00:00:00.276482'), 3.0)
-        dates = [np.datetime64('2016-01-01T11:59:59.861759000',),
-                 np.datetime64('2016-01-02T12:00:00.138241000')]
-        expected = Image((dates, [1.5, 2.5], [[1, 0], [0, 2]]),
-                         datatype=['xarray'], bounds=bounds, vdims=Dimension('Count', nodata=0))
-        expected2 = Image((dates, [1.5, 2.5], [[0, 1], [1, 1]]),
-                         datatype=['xarray'], bounds=bounds, vdims=Dimension('Count', nodata=0))
-        self.assertEqual(imgs[0], expected)
-        self.assertEqual(imgs[1], expected2)
+        assert isinstance(img, ImageStack)
+        x = pd.date_range(xstart, xend, periods=4).to_numpy()
+        y = np.array([1.25, 1.75, 2.25, 2.75])
+        cat = np.array([0.0, 1.0])
+        a = np.array([
+            [[1, 0],
+            [0, 0],
+            [0, 0],
+            [0, 1]],
+            [[0, 0],
+            [1, 0],
+            [0, 1],
+            [0, 0]],
+            [[0, 0],
+            [0, 1],
+            [1, 1],
+            [0, 0]],
+            [[0, 1],
+            [0, 0],
+            [0, 0],
+            [1, 0]]
+        ], dtype=np.uint32)
+        xrds = xr.DataArray(
+            a,
+            dims=('y', 'x', 'Cat'),
+            coords={"y": y, "Cat": cat, "x": x}
+        )
+        expected = ImageStack(xrds, kdims=["x", "y"], vdims=["0.0", "1.0"])
+        assert (expected.data == a).all()
```

Comment:
The fact that there's no registered `compare_x` testing function for ImageStack kept me busy for a little while, until I saw other ImageStack tests don't use `self.assertEqual()`.

Reply:
Yes. At some point, I plan to switch from unittest to pytest. To avoid adding extra work in the future, I try to avoid using `self.assertEqual` for new tests.

Reply:
Ok get that. It's just it'd be nice to have a `compare_imagestack` function somewhere that can be used in these cases.

Reply:
Just was hit by this myself... I wanted to look up this issue to write that we should add it. So thank you for the notification :100: